### PR TITLE
feat(app): add module attention warning to labware setup 

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

This PR addresses the UI tweaks + module attention warning outlined in #8212

# Changelog

- Add module attention warning to labware setup

# Review requests

See ACs in #8212

# Risk assessment

Low

### CODE REVIEW

File: app/src/organisms/ProtocolSetup/LabwareSetup/SecureLabwareModal.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+): JSX.Element => {
+  const { t } = useTranslation(['protocol_setup', 'shared'])
+  const moduleName = getModuleName(props.type)
+  return (
+    <Portal level="top">
+      <Modal className={styles.modal} contentsClassName={styles.modal_contents}>
+        <Box>
+          <Text as={'h3'} marginBottom={SPACING_3}>
+            {t(`secure_labware_modal_title`, {
+              name: moduleName,
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/LabwareSetup/SecureLabwareModal.tsx]: I tested a protocol with a TC and the name on the warning says "Magnetic Module"

Screen Shot 2021-08-11 at 12 53 57

Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwareSetup/SecureLabwareModal.tsx]: sorry that was outdated, if you pull it should be fixed

File: app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  const moduleName = getModuleName(props.moduleType)
+  return (
+    <Box>
+      <Text
+        as="h5"
+        marginTop={SPACING_2}
+        marginBottom={SPACING_1}
+        marginX={SPACING_2}
+      >
+        {getModuleName(props.moduleType)}
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx]: looks like you've already got this in a local var `moduleName`

Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx]: do i? i don't think it has been pulled out of props yet

Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx]: i can tho! that's usually the convention we follow anyways

File: app/src/assets/localization/en/protocol_setup.json [hunk 3]
```diff
@@ -10,6 +10,11 @@
   "labware_help_example": "<block>Alice is using a labware in Slot 6. During labware position check, she adjust the pipette position while checking the A1 of the labware to 0.2mm in X, and 1.2mm in Z. Later, Bob is preparing to run the same protocol on that robot. The labware offset that Alice created in Slot 6 will be applied for Bob’s protocol unless he changes or clears it.</block>",
   "offset_title": "Offset",
   "labware_position_check_text": "Labware Position Check is an optional workflow that guides you through checking the position of each labware on the deck. During this check, you can make an offset adjustment to the overall position of the labware.",
+  "magnetic_module_attention_warning": "<block>Opentrons recommends securing labware with the ”module’s bracket. <a>Securing labware to the Magnetic Module</a></block>",
```
Comment (jerader) [inline app/src/assets/localization/en/protocol_setup.json]: there's a random quotation here `”module’s bracket.`

Comment (shlokamin) [reply app/src/assets/localization/en/protocol_setup.json]: lol good catch idk how that ended up there

**Review by emilywools — APPROVED** [review]
team worrrrrrk!!! looks GTM!!!! thank you Shlok!!

**Review by jerader — APPROVED** [review]
Looking good now, no more rogue periods or quotations 🍾

File: app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx (hunk 2 above)
Comment (b-cooper) [reply app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx]: oh I meant line 34 in this file `const moduleName = getModuleName(props.moduleType)`
so you don't have to call `getModuleName` twice in the same scope right?

Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx]: lol yes good call i missed that, thanks!

File: app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        backgroundColor={COLOR_WARNING_LIGHT}
+        color={C_DARK_GRAY}
+      >
+        <Box margin={SPACING_3}>
+          <Flex margin={SPACING_2}>
+            <Box size={SIZE_2} paddingY={SPACING_1} paddingRight={SPACING_2}>
+              <Icon name="alert-circle" color={COLOR_WARNING} />
+            </Box>
+            <Text as="h4" marginY={SPACING_2}>
+              Some labware and modules require extra attention
```
Comment (shlokamin) [inline app/src/organisms/ProtocolSetup/LabwareSetup/ExtraAttentionWarning.tsx]: oop this should be in i18n

File: app/src/organisms/ProtocolSetup/LabwareSetup/SecureLabwareModal.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                    />
+                  ),
+                }}
+              />
+              <Flex justifyContent={JUSTIFY_CENTER} marginY={SPACING_3}>
+                <img src={secureMagModBracketImage} />
+              </Flex>
+              <Flex justifyContent={JUSTIFY_CENTER}>
+                <Text fontSize={FONT_SIZE_BODY_2} marginX={SPACING_4}>
+                  Magnetic Module with 0.2 mL 96 well PCR Plate
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/LabwareSetup/SecureLabwareModal.tsx]: a couple more pieces of copy here that slipped through the i18n sensors

File: app/src/organisms/ProtocolSetup/LabwareSetup/index.tsx [hunk 6]
```diff
@@ -78,15 +85,21 @@ export const LabwareSetup = (props: LabwareSetupProps): JSX.Element | null => {
...
           fontSize={FONT_SIZE_BODY_1}
           color={C_BLUE}
           alignSelf={ALIGN_FLEX_END}
           onClick={() => setShowLabwareHelpModal(true)}
           data-test={'LabwareSetup_helpLink'}
         >
           {t('labware_help_link_title')}
-        </Link>
+        </Btn>
+        {moduleTypesThatRequireExtraAtention.length > 0 && (
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/LabwareSetup/index.tsx]: *Attention

Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwareSetup/index.tsx]: doh, good catch

**Review by b-cooper — APPROVED** [review]
Everything looks good overall 🌊.

I dropped a couple of small comments. Also, the Labware Help Modal has a bit of extra y scroll when the view port is the default size, this seems avoidable. Both The Labware Help modal and the Extra Attention modal designs include a close "X" button in the top right of the modal in addition to the "Close" CTA button at the bottom, though they code doesn't include them atm, was that intentional?

**Review by b-cooper — APPROVED** [review]
🌵

**Pull request merged**